lib/streamlit/elements/image.py

```diff
@@ -50,12 +52,14 @@
 # DPI.
 MAXIMUM_CONTENT_WIDTH: Final[int] = 2 * 730
 
-PILImage: TypeAlias = Union[ImageFile.ImageFile, Image.Image]
+PILImage: TypeAlias = Union[
+    ImageFile.ImageFile, Image.Image, GifImagePlugin.GifImageFile
+]
 AtomicImage: TypeAlias = Union[PILImage, "npt.NDArray[Any]", io.BytesIO, str]
 ImageOrImageList: TypeAlias = Union[AtomicImage, List[AtomicImage]]
 UseColumnWith: TypeAlias = Optional[Union[Literal["auto", "always", "never"], bool]]
 Channels: TypeAlias = Literal["RGB", "BGR"]
-ImageFormat: TypeAlias = Literal["JPEG", "PNG"]
+ImageFormat: TypeAlias = Literal["JPEG", "PNG", "GIF"]
```

Comment:
Since this also adds `gif` as a supported argument to `output_format` we might need to add it to the docstring as well. Can I load a `png` image and specify `gif` as output format and it transforms it into a `gif`?

Reply:
That's a good point! I added the "GIF" type to `ImageFormat` only to avoid:

```sh
lib/streamlit/elements/image.py: note: In function "_validate_image_format_string":
lib/streamlit/elements/image.py:202:16: error: Incompatible return value type (got "Literal['GIF']", expected "Literal['JPEG', 'PNG']")  [return-value]
```

If you load any non-gif image (png, jpg, svg) and specify `output_format="GIF"`, it is not transformed into a gif. It follows the st.image logic and returns the same type as input or "PNG" or "JPG".

Reply:
Alright, probably we don't need to add any warning here that the argument is ignored since we don't do this for other situations as well (e.g. using an SVG as input with output_format=PNG).